README.md

```diff
@@ -160,9 +160,87 @@ Sends an HTTP POST request to the given URL (with Content-Type `application/x-ww
 
 Args:
 - url (the URL to send the request to)
-- element (a dot-delimited string representing the specific JSON element to return from the response body, or an empty string for the whole response body)
+- element (see section on "JSON Query Syntax" below)
 - string(s)... (any amount of strings, from previously set headers or literal values, which will be concatenated and delimited with '&' for the request body)
 
+##### JSON Query Syntax
+The element to return from the JSON response body is specified by a dot-delimited string representing the path to the element. Each part of this string is a single step in that path, and array access is handled by providing a zero-based index. If the element selected has a primitive value, that value is returned. If the element selected is a JSON object or array, that object/array is returned in compact form (insignificant whitespace removed). To return the entire JSON response body, use an empty string.
+
+**Examples:**  
+**Primitive data**  
+JSON:
+```json
+{
+    "nested": {
+        "object": {
+            "data": 123
+        }
+    }
+}
+```
+Query: 'nested.object.data'  
+Result: 123
+
+**Object**  
+JSON:
+```json
+{
+    "nested": {
+        "object": {
+            "data": 123
+        }
+    }
+}
+```
+Query: 'nested.object'  
+Result: {"data":123}
+
+**Array**  
+JSON:
+```json
+{
+    "nested": {
+        "array": [
+            "abc",
+            "def",
+            "ghi"
+        ]
+    }
+}
+```
+Query: 'nested.array'  
+Result: ["abc","def","ghi"]
+
+**Array element**  
+JSON:
+```json
+{
+    "nested": {
+        "array": [
+            "abc",
+            "def",
+            "ghi"
+        ]
+    }
+}
+```
+Query: 'nested.array.1'  
+Result: def
+
+**Whole response**  
```

Comment:
For this example, I would give a response that's not JSON to demonstrate it's use case for when you can't parse the response (because it's not JSON).